internal/federationdomain/oidc/oidc.go

```diff
@@ -104,26 +105,152 @@ type UpstreamStateParamData struct {
 	FormatVersion string              `json:"v"`
 }
 
-// Get the defaults for the Supervisor server.
+// DefaultOIDCTimeoutsConfiguration returns the default timeouts for the Supervisor server.
 func DefaultOIDCTimeoutsConfiguration() timeouts.Configuration {
-	accessTokenLifespan := 2 * time.Minute
+	// Note: The maximum time that users can access Kubernetes clusters without
+	// needing to do a Supervisor refresh is the sum of the access token lifetime,
+	// the ID token lifetime, and the Concierge's mTLS client cert lifetime.
+	// This is because a client can exchange the access token just before it expires
+	// for a new cluster-scoped ID token, and use that just before it expires to get
+	// a new mTLS client cert, which grants access to the cluster until it expires.
+	//
+	// Note that the Concierge's mTLS client cert lifetime is 5 minutes, which can
+	// be seen in its source at credentialrequest/rest.go.
+	//
+	// This maximum total time is important because it represents the longest possible
+	// time that a user could continue to use a cluster based on their original login
+	// (or most recent refresh) after an administrator of an external identity provider
+	// removes the user, revokes their session, changes their group membership,
+	// or otherwise makes any type of change to the user's account in the external
+	// identity provider that should be noticed by the Supervisor during an upstream
+	// refresh.
+	//
+	// Given the timeouts specified below:
+	// For sessions started using a GitHub PAT, this is 8 + 2 + 5 = 15 minutes.
```

Comment:
So here was want a possible additional 6 minutes, not a huge amount.  Your description on the PR itself gives this extra `This would allow us to avoid storing the PAT in the Supervisor's session storage for use during refresh.`.  

I'm still not 100% sure I understand why the additional 6 minutes is worth the extra bits of code, given its still relatively short.  🤔

Reply:
This section of the Github proposal elaborates: https://github.com/vmware-tanzu/pinniped/pull/1860/files#diff-99eac6e887641e8a0eaf30490a8ef2887752dca568db541a1f47d8c01a596495R321

Reply:
Yes, as described in the proposal doc, making a small change to the lifetime can make a big difference overall, especially when there are lots of simultaneous users.